rest_framework_swagger/tests.py

```diff
@@ -651,6 +651,49 @@ def post(self, request, *args, **kwargs):
         self.assertNotIn('parameters', post)
         self.assertIn('--iron-socks', post['notes'])
 
+    def get_introspector_test(self):
+        """
+        yaml method specifications are dependent on all patterns
+        """
+        class MyViewSet(ModelViewSet):
+            """
+            ---
+            list:
+                param: my param
+            retrieve:
+                param: my param
+            """
+            serializer_class = CommentSerializer
+            model = User
+
+        # Test an object endpoint
+        url_patterns = patterns(
+            '',
+            url(
+                r'^/api/endpoint/(?P<pk>[^/]+)$',
+                MyViewSet.as_view({
+                    'get': 'retrieve',
+                    'put': 'update',
+                    'patch': 'partial_update',
+                    'delete': 'destroy'
+                })
+            ),
+            url(
+                r'^/api/endpoint/$',
+                MyViewSet.as_view({
+                    'get': 'list',
+                    'post': 'create',
+                }))
+        )
+        urlparser = UrlParser()
+        apis = urlparser.get_apis(url_patterns)
+        api = apis[0]
+        self.assertIn('pk', api['pattern'].regex.pattern)
+        generator = DocumentationGenerator()
+        introspector = generator.get_introspector(api, apis)
```

Comment:
The issue here is the two methods listed in the docstring - list and retrieve. They are provided by the two patterns, and if the introspector doesn't have access to all the patterns, on each call to get_introspector it will complain that one or the other is missing from the list of methods that the view supports.